SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -286,6 +286,11 @@ - (void)enqueueMessage:(TSOutgoingMessage *)message
         OWSAssert([message.body lengthOfBytesUsingEncoding:NSUTF8StringEncoding] <= kOversizeTextMessageSizeThreshold);
     }
 
+    if (message.isExpiringMessage && message.quotedMessage != nil) {
+        failureHandler(OWSErrorMakeAssertionError(@"expiring message unexpectedly had a quotedMessage"));
```

Comment:
I don't think this part is right.

* Alice talking to Bob.
* Conversation has disappearing messages off.
* Alice says "hi".
* Bob turns disappearing messages on.
* Bob responds to "hi" with "hello!".  

It's okay for "hello" (expiring) to quote "hi" (non-expiring).

Reply:
good catch, fixed.